Tests/SonarScanner.MSBuild.PreProcessor.Test/TargetsInstallerTests.cs

```diff
@@ -25,403 +25,377 @@
-            }
-            else
-            {
-                // Copy is not executed
-                this.fileWrapperMock.Verify(x => x.Copy(sourcePath, "c:\\project\\bin\\targets\\SonarQube.Integration.targets", It.IsAny<bool>()), Times.Never);
-            }
+    private void InstallLoaderTargets_InternalCopyTargetFileToProject(string sourceContent, bool destinationExists, string destinationContent = null)
+    {
+        // Arrange
+        var targetsInstaller = new TargetsInstaller(logger, msBuildPathSettingsMock, fileWrapperMock, directoryWrapperMock);
+        var sourcePath = Path.Combine(
+            Path.GetDirectoryName(typeof(ArgumentProcessor).Assembly.Location),
+            "Targets",
+            "SonarQube.Integration.targets");
+        fileWrapperMock.ReadAllText(sourcePath).Returns(sourceContent);
+        fileWrapperMock.ReadAllText("c:\\project\\bin\\targets\\SonarQube.Integration.targets").Returns(destinationContent);
+        fileWrapperMock.Exists("c:\\project\\bin\\targets\\SonarQube.Integration.targets").Returns(destinationExists);
+
+        // Act
+        using (new AssertIgnoreScope())
+        {
+            targetsInstaller.InstallLoaderTargets("c:\\project");
         }
 
-        private void InstallLoaderTargets_InternalCopyTargetsFile(string sourceContent, bool destinationExists, string destinationContent = null)
+        // Assert
+        var sameContent = sourceContent.Equals(destinationContent);
+        if (!destinationExists || !sameContent)
         {
-            // Arrange
-            var targetsInstaller = new TargetsInstaller(this.logger, this.msBuildPathSettingsMock.Object, this.fileWrapperMock.Object, this.directoryWrapperMock.Object);
-            var sourcePath = Path.Combine(
-                Path.GetDirectoryName(typeof(ArgumentProcessor).Assembly.Location),
-                "Targets",
-                "SonarQube.Integration.ImportBefore.targets");
-
-            this.msBuildPathSettingsMock
-                .Setup(x => x.GetImportBeforePaths())
-                .Returns(new[] { "c:\\global paths" });
-            this.fileWrapperMock
-                .Setup(x => x.ReadAllText(sourcePath))
-                .Returns(sourceContent);
-            this.fileWrapperMock
-                .Setup(x => x.ReadAllText("c:\\global paths\\SonarQube.Integration.ImportBefore.targets"))
-                .Returns(destinationContent);
-            this.fileWrapperMock
-                .Setup(x => x.Exists("c:\\global paths\\SonarQube.Integration.ImportBefore.targets"))
-                .Returns(destinationExists);
-
-            // Act
-            using (new AssertIgnoreScope())
-            {
-                targetsInstaller.InstallLoaderTargets("c:\\project");
-            }
+            // Copy is executed once, overwriting existing files
+            fileWrapperMock.Received(1).Copy(sourcePath, "c:\\project\\bin\\targets\\SonarQube.Integration.targets", true);
```

Comment:
Reliability improvement: The string should be extracted as a constant inside the method, to prevent accidental typos. While you're on it, you can make it verbatim to get rid of the `\\`.

While it is safe here and above, it is risky below in the `DidNotReceive`